### PR TITLE
Add tests / generate FFI bindings

### PR DESCRIPTION
**Pull request by HeinrichHartmann**

### CODE REVIEW

File: fqs.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*
+ * fqs
+ *
+ * Usage:
+ *
+ *     fqs <host> <port> <user> <pass> <exchange> <route>
```
Comment (postwait) [inline fqs.c]: If we're going to install this for users, let's `getopt` it.

**Comment by HeinrichHartmann** [comment]
Added getopt + test.

File: fqs.c [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  printf("\t-a <host>:<port>\tspecify the address to connect to (default: 127.0.0.1:8765)\n");
+  printf("\t-x <exchange>\tspecify the exchange to relay messages on (required)\n");
+  printf("\t-r <route>\tspecify the message route (required)\n");
+  printf("\t-u <user>\tspecify the user (default: user)\n");
+  printf("\t-p <pass>\tspecify the password (default: pass)\n");
+}
+
+int main(int argc, char **argv) {
+  char *host = NULL, *exchange = NULL, *route = NULL;
+  char *address = strdup("127.0.0.1:8765");
```
[the remark this answers is not in the document]
Comment (postwait) [reply fqs.c]: you can mutate things on the stack... it's not declared const.

It's just a nit.  The program doesn't leak as these are allocated an used in main once, so they free on exit.  I approved.

[the remark this answers is not in the document]
Comment (HeinrichHartmann) [reply fqs.c]: I am pretty sure I tried that and it segfaulted. If I understand this correctly the pointer is on the stack but the actual string is "interned" in the data segment, which is read-only.

I'll try this again.

**Comment by HeinrichHartmann** [comment]
Cleaned up history.

**Pull request merged**